tests/test_download.py

```diff
@@ -168,81 +170,148 @@ def test_download_url_to_tempfileobj_and_urls(self):
     self.assertRaises(securesystemslib.exceptions.FormatError,
                       download_file, None, self.target_data_length)
 
-    self.assertRaises(securesystemslib.exceptions.FormatError,
+    self.assertRaises(tuf.exceptions.URLParsingError,
                       download_file,
                       self.random_string(), self.target_data_length)
 
-    self.assertRaises(six.moves.urllib.error.HTTPError,
+    self.assertRaises(requests.exceptions.HTTPError,
                       download_file,
                       'http://localhost:' + str(self.PORT) + '/' + self.random_string(),
                       self.target_data_length)
 
-    self.assertRaises(six.moves.urllib.error.URLError,
+    self.assertRaises(requests.exceptions.ConnectionError,
                       download_file,
                       'http://localhost:' + str(self.PORT+1) + '/' + self.random_string(),
                       self.target_data_length)
 
     # Specify an unsupported URI scheme.
     url_with_unsupported_uri = self.url.replace('http', 'file')
-    self.assertRaises(securesystemslib.exceptions.FormatError, download_file, url_with_unsupported_uri,
+    self.assertRaises(requests.exceptions.InvalidSchema, download_file, url_with_unsupported_uri,
                       self.target_data_length)
-    self.assertRaises(securesystemslib.exceptions.FormatError, unsafe_download_file,
+    self.assertRaises(requests.exceptions.InvalidSchema, unsafe_download_file,
                       url_with_unsupported_uri, self.target_data_length)
 
 
 
-  def test__get_opener(self):
-    # Test normal case.
-    # A simple https server should be used to test the rest of the optional
-    # ssl-related functions of 'tuf.download.py'.
-    fake_cacert = self.make_temp_data_file()
 
-    with open(fake_cacert, 'wt') as file_object:
-      file_object.write('fake cacert')
 
-    tuf.settings.ssl_certificates = fake_cacert
-    tuf.download._get_opener('https')
-    tuf.settings.ssl_certificates = None
+  # This test uses sites on the internet, requiring a net connection to succeed.
+  # Since this is the only such test in TUF, I'm not going to enable it... but
```

Comment:
I think we should turn this on, they are very useful tests, and these days it's reasonable to assume tests can reach out to the Internet...

Reply:
I added a local test for the wrong host scenario, and I've just now added a local test for the expired cert scenario. That should suffice. It's still nice to have the online tests, but I'd still rather keep them off and leave them as diagnostic tools as long as they aren't much more effective; net connection issues causing test failures aren't worth it yet.

Reply:
(You could make turning on the tests easier though --- wrap them in `"""` instead of per-line `# ` prefixing.)

Reply:
Good tip. Has to be `'''`, since the docstring inside uses """.